### PR TITLE
Issue #156 Add codefix for S2328 (GetHashCode uses mutable field)

### LINKED ISSUE #156: Add codefix for S2328 (GetHashCode uses mutable field)

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #156

### CODE REVIEW

File: src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            return hash;
+        }
+        public int SomeMethod()
+        {
+            int hash = Zero;
+            hash += this.age.GetHashCode();
+            return hash;
+        }
+    }
+}
```
Comment (duncanpMS) [inline src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs]: Additional test case:
* int foo, bar; // can't make readonly in this case

File: src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            hash += age.GetHashCode(); // Noncompliant
+            hash += this.name.GetHashCode(); // Noncompliant
+            hash += this.birthday.GetHashCode();
+            hash += Field; // Noncompliant
+            return hash;
+        }
+        public int SomeMethod()
+        {
+            int hash = Zero;
+            hash += this.age.GetHashCode();
```
Comment (duncanpMS) [inline src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs]: Adding readonly will make the code uncompilable if the variable is assigned to other than in the constructor. However, that's probably ok; the user can preview the change, and can easily undo the change.

**Comment by duncanpMS** [comment]
LGTM

File: src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public class GetHashCodeMutable : AnyOther
+    {
+        public readonly DateTime birthday;
+        public const int Zero = 0;
+        public readonly int age;
+        public readonly string name;
+
+        public GetHashCodeMutable()
+        {
+            any = new AnyOther();
```
Comment (duncanpMS) [inline src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs]: *any* isn't declared.

File: src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs (hunk 2 above)
Comment (tvsonar) [inline src/Tests/SonarLint.UnitTest/TestCases/GetHashCodeMutable.Fixed.cs]: @duncanpMS This is expected. We never want to produce code that is semantically wrong, but compiles. Also, it would be very-very nice to produce code that is semantically correct and compiles, but at the moment for this rule this is not always the case. There is a rule to check if a field can be marked readonly, whenever we decide how to handle rules that depend on each other, this issue will be fixed. I've created #458 to note this issue.
